javagen/src/main/java/com/azure/autorest/template/StreamSerializationModelTemplate.java

```diff
@@ -295,8 +295,14 @@ private static void serializeJsonProperties(JavaBlock methodBlock, ClientModelPr
             methodBlock.ifBlock(additionalPropertiesAccessExpr + " != null", ifAction -> {
                 IType valueType = ((MapType) wireType).getValueType().asNullable();
                 ifAction.line("for (Map.Entry<String, %s> additionalProperty : %s.entrySet()) {", valueType, additionalPropertiesAccessExpr);
-                ifAction.indent(() ->
-                        ifAction.line("jsonWriter.writeUntypedField(additionalProperty.getKey(), additionalProperty.getValue());"));
+                ifAction.indent(() -> {
+                    if (valueType == ClassType.BINARY_DATA) {
+                        // Special handling for BinaryData
+                        ifAction.line("jsonWriter.writeUntypedField(additionalProperty.getKey(), additionalProperty.getValue() == null ? null : additionalProperty.getValue().toObject(Object.class));");
```

Comment:
why do we have  `.toObject(Object.class)` here?

Reply:
`writeUntypedField` does not work on `BinaryData`. So it had to be converted to `Object` first.

This is already done in defined properties. Just it missing in `additionalProperties`.

Reply:
Also this is my comment of "this is not the most efficient runtime code, but this is the code that works".

`BinaryData` should already contain serialized code. Here we deserialize it and let the `writeUntypedField` serialize that Object.

If possible the optimal procedure should be a direct "merge" of what in `BinaryData` and the rest of the model, but that may not have support in azure-json.